liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/applications/InstallAppMojoSupport.java

```diff
@@ -54,18 +74,136 @@ protected void installApp(Artifact artifact) throws Exception {
+            file = file.substring(file.lastIndexOf(File.separator) + 1);
+            copyFile.setTofile(new File(destDir, file));
         } else {
             copyFile.setTodir(destDir);
         }
         copyFile.execute();
     }
+    
+    // install project artifact using looseconfig file 
+    protected void installLooseConfigApp() throws Exception {
+        //Artifact artifact = project.getArtifact();
+        File destDir = new File(serverDirectory, appsDirectory);
+        File looseConfigFile = new File(destDir, getLooseConfigFileName(project));
+        LooseConfigData config = new LooseConfigData();
+        
+        log.info(MessageFormat.format(messages.getString("info.install.app"), getLooseConfigFileName(project)));
+        
+        File dir = new File(project.getBasedir() + "/src/main/webapp");
+        if (dir.exists()) {
+            config.addDir(project.getBasedir() + "/src/main/webapp", "/");
+        }
+        
+        dir = new File(project.getBuild().getOutputDirectory());
+        if (dir.exists()) {
+            config.addDir(project.getBuild().getOutputDirectory(), "/WEB-INF/classes");
+        } else {
+            // TODO: need to revise the message
+            throw new MojoExecutionException(
+                    "Project has not be compiled yet. Please run mvn install or mvn compile to build the project first.");
+        }
+        
+        // retrieves dependent library jar files
+        List<Artifact> libraries = getDependentLibraries();
+        if (!libraries.isEmpty()) {
+            // get a list of dependent-modules from eclipse project deployment
+            // assembly if running in eclipse
+            List<String> eclipseModules = getEclipseDependentMods();
 
+            // referencing dependent library jar file from mvn repository or set
+            // loose config reference to dependent eclipse project output classpath
+            if (eclipseModules.isEmpty()) {
+                addLibraryFromM2(libraries, config);
+            } else {
+                for (Artifact library : libraries) {
+                    if (library.getFile() != null && eclipseModules.contains(library.getFile().getName())) {
+                        MavenProject module = getSiblingModule(library);
+                        config.addArchive(module.getBuild().getOutputDirectory(),
+                                "/WEB-INF/lib/" + module.getBuild().getFinalName() + "." + module.getPackaging());
+                    } else {
+                        addLibraryFromM2(library, config);
+                    }
+                }
+            }
+        }
+        
+        config.toXmlFile(looseConfigFile);
+    }
+    
+    // get loose configuration file name for project artifact
+    private String getLooseConfigFileName(MavenProject project) {
+        String name = project.getBuild().getFinalName() + "." + project.getPackaging();
```

Comment:
This feels like the bug we had before when the packaging type was liberty-assembly and war file got named wrong.